### ISSUE TITLE
Compute GEN-level free energy results

### ISSUE THREAD

**Issue by mcwitt** [choderalab/covid-moonshot-infra#35]
https://github.com/choderalab/covid-moonshot-infra/pull/34#discussion_r477653314

**Comment by jchodera** [comment]
If we organize all the work values in a flat array, we can probably just have another flat `int` array that annotates which GEN each work value came from, e.g.
```JSON
"work_forward": [1.234, 2.345, ...]
"work_reverse": [4.523, -1.235, ...]
"work_gen": [0, 0, 1, ...]
```

**Comment by mcwitt** [comment]
Makes sense, and then we could record the free energies by gen with something like
```json
"delta_f": [1.234, 2.345, ...],
"ddelta_f": [0.12, 0.23, ...]
```
where there is one entry per gen.

In the PR I had started down the path of implementing a more nested structure, which looks like
```json
{
  "analysis": {
    "complex_phase": {
      "fe": {
        "delta_f": 42.42,
        "ddelta_f": 4.24,
        "bar_overlap": 0.85,
        "num_work_values": 42
      },
      "gens": [
        {
          "fe": {
            "delta_f": 42.42,
            "ddelta_f": 4.24,
            "bar_overlap": 0.85,
            "num_work_values": 42
          },
          "forward_works": [42.4, 40.34, 44.83],
          "reverse_works": [40.55, 38.12, 37.2]
        }
      ]
    }
  }
}
```
I like this from the perspective of "making invalid states unrepresentable", but admittedly the nesting might be more annoying to deal with downstream.

**Comment by jchodera** [comment]
Nesting shouldn't be a problem!

But let's use more descriptive, self-documenting names if possible. "fe" is a little too short to be unambiguous.

Eventually, we can overhaul this so that even the units are self-documenting, either `forward_works_in_kT` or using a units library like [pint](https://pint.readthedocs.io/en/stable/) or [unyt](https://unyt.readthedocs.io/en/stable/).

**Comment by mcwitt** [comment]
Implemented in #36
